File: company-service/src/main/java/ru/panic/companyservice/template/repository/impl/CompanyRepositoryImpl.java
package ru.panic.companyservice.template.repository.impl;

import org.jooq.DSLContext;
import org.springframework.stereotype.Repository;
import ru.panic.companyservice.generatedClasses.tables.CompaniesTable;
import ru.panic.companyservice.template.entity.Company;
import ru.panic.companyservice.template.repository.CompanyRepository;

import java.util.List;

@Repository
public class CompanyRepositoryImpl implements CompanyRepository {
    public CompanyRepositoryImpl(DSLContext dslContext) {
        this.dslContext = dslContext;
    }

    private final DSLContext dslContext;

    @Override
    public void save(Company company) {
        dslContext.insertInto(CompaniesTable.COMPANIES_TABLE)
                .set(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME, company.getCompanyName())
                .set(CompaniesTable.COMPANIES_TABLE.OWNER, company.getOwner())
                .set(CompaniesTable.COMPANIES_TABLE.WEBHOOK_URL, company.getWebhookUrl())
                .set(CompaniesTable.COMPANIES_TABLE.SECRET_KEY, company.getSecretKey())
                .set(CompaniesTable.COMPANIES_TABLE.CREATED_AT, company.getCreatedAt())
                .execute();
    }

    @Override
    public void delete(Company company) {
        dslContext.deleteFrom(CompaniesTable.COMPANIES_TABLE)
                .where(CompaniesTable.COMPANIES_TABLE.ID.eq(company.getId()))
                .and(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME.eq(company.getCompanyName()))
                .and(CompaniesTable.COMPANIES_TABLE.OWNER.eq(company.getOwner()))
                .and(CompaniesTable.COMPANIES_TABLE.WEBHOOK_URL.eq(company.getWebhookUrl()))
                .and(CompaniesTable.COMPANIES_TABLE.SECRET_KEY.eq(company.getSecretKey()))
                .and(CompaniesTable.COMPANIES_TABLE.CREATED_AT.eq(company.getCreatedAt()))
                .execute();
    }

    @Override
    public List<Company> findAllCompanyByOwner(String owner) {
        return dslContext.selectFrom(CompaniesTable.COMPANIES_TABLE)
                .where(CompaniesTable.COMPANIES_TABLE.OWNER.eq(owner))
                .fetchInto(Company.class);
    }

    @Override
    public boolean extendedByCompanyName(String companyName) {
        Company company = dslContext.selectFrom(CompaniesTable.COMPANIES_TABLE)
                .where(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME.eq(companyName))
                .fetchOneInto(Company.class);
        return company != null;
    }
}

File: company-service/src/main/kotlin/ru/panic/companyservice/template/repository/impl/CompanyRepositoryImpl.kt
package ru.panic.companyservice.template.repository.impl

import org.jooq.DSLContext
import org.springframework.stereotype.Repository
import ru.panic.companyservice.generatedClasses.tables.CompaniesTable
import ru.panic.companyservice.template.entity.Company
import ru.panic.companyservice.template.repository.CompanyRepository
@Repository
class CompanyRepositoryImpl(private val dslContext: DSLContext) : CompanyRepository {
    override fun save(company: Company) {
        dslContext.insertInto(CompaniesTable.COMPANIES_TABLE)
            .set(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME, company.companyName)
            .set(CompaniesTable.COMPANIES_TABLE.OWNER, company.owner)
            .set(CompaniesTable.COMPANIES_TABLE.WEBHOOK_URL, company.webhookUrl)
            .set(CompaniesTable.COMPANIES_TABLE.SECRET_KEY, company.secretKey)
            .set(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME, company.companyName)
            .execute()
    }
    override fun findCompanyByCompanyName(companyName: String): Company? {
        return dslContext.selectFrom(CompaniesTable.COMPANIES_TABLE)
            .where(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME.eq(companyName))
            .fetchOneInto(Company::class.java)
    }

    override fun deleteByCompanyName(companyName: String) {
        dslContext.deleteFrom(CompaniesTable.COMPANIES_TABLE)
            .where(CompaniesTable.COMPANIES_TABLE.COMPANY_NAME.eq(companyName))
            .execute()
    }

    override fun extendedBySecretKey(secretKey: String): Boolean {
        val company: List<Company?> = dslContext.selectFrom(CompaniesTable.COMPANIES_TABLE)
            .where(CompaniesTable.COMPANIES_TABLE.SECRET_KEY.eq(secretKey))
            .fetchInto(Company::class.java)
        return company.isEmpty()
    }

    override fun findAllCompanyByOwner(owner: String): List<Company>? {
        return dslContext.selectFrom(CompaniesTable.COMPANIES_TABLE)
            .where(CompaniesTable.COMPANIES_TABLE.OWNER.eq(owner))
            .fetchInto(Company::class.java)
    }
}